### PR TITLE
Move demo to Docker Compose, tidy up bugs and docs on the way

### PR DESCRIPTION
**Pull request by rmoff**
Existing Clickstream demo is based on single big docker image, which makes it more difficult to test as part of the release cycle, and more difficult to work with in general. 

This PR splits it up into Docker Compose, as well as fixing some bugs along the way and tweaking the documentation for clarity.

### CODE REVIEW

**Comment by rmoff** [comment]
Reviews please :) 

@confluentinc/ksql this was originally developed by you folk, so need eyes on please
@JimGalasyn this is changing the docs - would be good if you can run the new demo through and confirm that it works. 
@ybyzek this is a quickstart/tutorial so just pinging for info really

File: docs/tutorials/clickstream-docker.rst [hunk 1]
```diff
@@ -7,135 +7,108 @@ These steps will guide you through how to setup your environment and run the cli
...
+#. Navigate to the KSQL repository ``docs/tutorials/`` directory and launch the tutorial in
+   Docker. Depending on your network speed, this may take up to 5-10 minutes.
 
-    .. code:: bash
-
-        [ ok ] Starting Grafana Server:.
+   .. code:: bash
 
-#.  Start |cp|.
+       $ cd ksql-clickstream-demo
```
Comment (JimGalasyn) [inline docs/tutorials/clickstream-docker.rst]: Currently, the `ksql-clickstream-demo` directory is under the `ksql` root directory. I assume it should be in `docs/tutorials` instead.

File: docs/tutorials/clickstream-docker.rst [hunk 2]
```diff
@@ -315,55 +309,64 @@ Send the KSQL tables to Elasticsearch and Grafana.
         ksql>
         Exiting KSQL.
 
-2. Navigate to the tutorial directory in the Docker container:
+2. Set up the required Elasticsearch document mapping template
```
Comment (JimGalasyn) [inline docs/tutorials/clickstream-docker.rst]: I can't test any further on Ubuntu 18.04, since I don't have an Elasticsearch container.

**Comment by rmoff** [comment]
Thanks Jim - do you have the logs from the container with the error please?

**Comment by JimGalasyn** [comment]
@rmoff Here you go! [es-container-failure.log](https://github.com/confluentinc/ksql/files/2375987/es-container-failure.log)

File: docs/tutorials/clickstream-docker.rst (hunk 1 above)
Comment (rmoff) [reply docs/tutorials/clickstream-docker.rst]: Not sure what you're saying. It is under the root. Do you want it moved?

File: docs/tutorials/clickstream-docker.rst [hunk 3]
```diff
@@ -7,135 +7,108 @@ These steps will guide you through how to setup your environment and run the cli
...
-        Starting ksql-server
-        ksql-server is [UP]
-
-    .. tip:: If you receive an out of memory error, see the :ref:`prerequisites <ksql_clickstream-docker>`.
+        $ docker-compose ps
+                        Name                                Command               State                Ports
+        -------------------------------------------------------------------------------------------------------------------
+        ksql-clickstream-demo_control-center_1    /etc/confluent/docker/run        Up      0.0.0.0:9021->9021/tcp
+        ksql-clickstream-demo_datagen-ratings_1   bash -c echo Waiting for K ...   Up
+        ksql-clickstream-demo_elasticsearch_1     /usr/local/bin/docker-entr ...   Up      0.0.0.0:9200->9200/tcp, 9300/tcp
```
[the remark this answers is not in the document]
Comment (rmoff) [reply docs/tutorials/clickstream-docker.rst]: From the log you sent me: 

    max virtual memory areas vm.max_map_count [65530] is too low, increase to at least [262144]

I'll update the instructions to address this.

[the remark this answers is not in the document]
Comment (rmoff) [reply docs/tutorials/clickstream-docker.rst]: @JimGalasyn I've updated the pre-reqs to add a note re. setting `vm.max_map_count` - please can you try it out. Thanks.

File: docs/tutorials/clickstream-docker.rst (hunk 2 above)
Comment (rmoff) [reply docs/tutorials/clickstream-docker.rst]: did you get chance to re-test this after my update(https://github.com/confluentinc/ksql/pull/1875#discussion_r217345050) ?

**Comment by rmoff** [comment]
> The impact of this change is that we will stop publishing the ksql-clickstream-demo docker image from 5.0.1 onward. I assume that that is intentional?

Yes I think so - because it would be too much overhead to try and maintain both. Any objections?

File: docs/tutorials/clickstream-docker.rst (hunk 2 above)
Comment (JimGalasyn) [reply docs/tutorials/clickstream-docker.rst]: I did, and the Elasticsearch container runs now!

File: docs/tutorials/clickstream-docker.rst [hunk 4]
```diff
@@ -315,55 +317,64 @@ Send the KSQL tables to Elasticsearch and Grafana.
         ksql>
         Exiting KSQL.
 
-2. Navigate to the tutorial directory in the Docker container:
+2. Set up the required Elasticsearch document mapping template
 
    .. code:: bash
 
-       $ cd /usr/share/doc/ksql-clickstream-demo/
+       $ docker-compose exec elasticsearch bash -c '/scripts/elastic-dynamic-template.sh'
```
Comment (JimGalasyn) [inline docs/tutorials/clickstream-docker.rst]: On Ubuntu, this command gives me an error:
```
jim@Maeve:~/repos/ksql/ksql-clickstream-demo$ docker-compose exec elasticsearch bash -c '/scripts/elastic-dynamic-template.sh'

-> Removing kafkaconnect index (if it doesn't exist, this just throws 404)

{"error":{"root_cause":[{"type":"index_template_missing_exception","reason":"index_template [kafkaconnect] missing"}],"type":"index_template_missing_exception","reason":"index_template [kafkaconnect] missing"},"status":404}

-> Loading Elastic Dynamic Template to ensure _TS fields are used for TimeStamp


{"acknowledged":true}
```

Comment (rmoff) [reply docs/tutorials/clickstream-docker.rst]: that's the expected error that the text mentions (404). Should I reword it to make it clearer?

Comment (JimGalasyn) [reply docs/tutorials/clickstream-docker.rst]: I think it would be helpful if you mention it up front in the topic -- I almost gave up, because it looks like a completely fatal error, and "just throws 404" didn't give me confidence. ^_^

File: docs/tutorials/clickstream-docker.rst (hunk 1 above)
Comment (JimGalasyn) [reply docs/tutorials/clickstream-docker.rst]: Sorry I meant that the procedure should be, "Navigate to the KSQL repository `ksql-clickstream-demo` directory..."

File: docs/tutorials/clickstream-docker.rst [hunk 5]
```diff
@@ -7,135 +7,108 @@ These steps will guide you through how to setup your environment and run the cli
...
-    .. code:: bash
+#. Switch to the correct |cp| release branch:
 
-        $ /etc/init.d/grafana-server start
+   .. codewithvars:: bash
+   
+       $ git checkout |release_post_branch|
 
-    Your output should resemble:
+#. Navigate to the KSQL repository ``docs/tutorials/`` directory and launch the tutorial in
```
Comment (JimGalasyn) [inline docs/tutorials/clickstream-docker.rst]: I think this should be "Navigate to the KSQL repository `ksql-clickstream-demo` directory..."

File: docs/tutorials/clickstream-docker.rst (hunk 4 above)
Comment (rmoff) [reply docs/tutorials/clickstream-docker.rst]: Good point. I've updated the scripts to be much cleaner in their output.

**Comment by rmoff** [comment]
@confluentinc/ksql could I get a +1 review on this please? thanks.

File: ksql-clickstream-demo/demo/clickstream-schema.sql [hunk 6]
```diff
@@ -1,11 +1,7 @@
--- lets the windows accumulate more data
-set 'commit.interval.ms'='2000';
-set 'cache.max.bytes.buffering'='10000000';
-set 'auto.offset.reset'='earliest';
 
 
 -- 1. SOURCE of ClickStream
-DROP STREAM IF EXISTS clickstream;
+--DROP STREAM IF EXISTS clickstream;
```
Comment (big-andy-coates) [inline ksql-clickstream-demo/demo/clickstream-schema.sql]: Why are we removing these statements?

File: ksql-clickstream-demo/demo/clickstream-schema.sql [hunk 7]
```diff
@@ -1,11 +1,7 @@
--- lets the windows accumulate more data
-set 'commit.interval.ms'='2000';
```
Comment (big-andy-coates) [inline ksql-clickstream-demo/demo/clickstream-schema.sql]: Why are we removing these statements?

File: ksql-clickstream-demo/src/main/java/io/confluent/ksql/DemoTest.java [hunk 8]
```diff
@@ -1,24 +0,0 @@
...
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- **/
-
-package io.confluent.ksql;
-
-public class DemoTest {
```
Comment (big-andy-coates) [inline ksql-clickstream-demo/src/main/java/io/confluent/ksql/DemoTest.java]: You might find that this is what is causing your maven build to fail.  I'm guessing maven doesn't like having a module with no source.
